add botpress v12

maximilianoPizarro opened the pull request:
#### What is this PR About?
# Deploy Botpress Server Community Helm Charts on Red Hat OpenShift
<link rel="icon" href="https://raw.githubusercontent.com/maximilianoPizarro/botpress-helm-chart/main/favicon-152.ico" type="image/x-icon" >
<p align="left">
<img src="https://img.shields.io/badge/redhat-CC0000?style=for-the-badge&logo=redhat&logoColor=white" alt="Redhat">
<img src="https://img.shields.io/badge/kubernetes-%23326ce5.svg?style=for-the-badge&logo=kubernetes&logoColor=white" alt="kubernetes">
<img src="https://img.shields.io/badge/helm-0db7ed?style=for-the-badge&logo=helm&logoColor=white" alt="Helm">
<img src="https://img.shields.io/badge/shell_script-%23121011.svg?style=for-the-badge&logo=gnu-bash&logoColor=white" alt="shell">
<a href="https://www.linkedin.com/in/maximiliano-gregorio-pizarro-consultor-it"><img src="https://img.shields.io/badge/LinkedIn-0077B5?style=for-the-badge&logo=linkedin&logoColor=white" alt="linkedin" /></a>
<a href="https://artifacthub.io/packages/search?repo=botpress"><img src="https://img.shields.io/endpoint?url=https://artifacthub.io/badge/repository/botpress" alt="Artifact Hub" /></a>
</p>

[![Open](https://img.shields.io/static/v1?label=Open%20in&message=Developer%20Sandbox&logo=eclipseche&color=FDB940&labelColor=525C86)](https://workspaces.openshift.com/#https://github.com/maximilianoPizarro/botpress-helm-chart/tree/main?storageType=ephemeral)

Watch a demo video
[![Botpress on Red Hat Developer Sandbox](https://img.youtube.com/vi/DBwNvq491Jw/0.jpg)](https://www.youtube.com/watch?v=DBwNvq491Jw)


The purpose of this project is to generate the kubernetes objects based on the node image from the official repository <a href="https://botpress.com">botpress</a> for deployment on container platforms using the Helm Charts strategy. Verified in <a href="https://developers.redhat.com/developer-sandbox">Sandbox RedHat OpenShift Dedicated</a> (Openshift 4.14.1).

#### How do we test this?
# Installation

## Charts Values Parameters

| Parameter                     | Value                                   | Default                        |
|-----------------------------|-----------------------------------------------------|------------------------------------|
| env.EXTERNAL_URL              | https://my-botpress-NAMESPACE-dev.apps.sandbox-m2.ll9k.p1.openshiftapps.com                                               | https://botpress-server-maximilianopizarro5-dev.apps.sandbox-m2.ll9k.p1.openshiftapps.com               |
| route.host            | my-botpress-NAMESPACE-dev.apps.sandbox-m2.ll9k.p1.openshiftapps.com                                               | botpress-server-maximilianopizarro5-dev.apps.sandbox-m2.ll9k.p1.openshiftapps.com |
| image.repository        | botpress/server                                               | quay.io/maximilianopizarro/botpress-server-v12         |


## Add repository

```bash
helm repo add botpress https://redhat-cop.github.io/helm-charts
```

## Install Chart with parameters

```bash
helm install botpress botpress/botpress --version VERSION --set route.host="Your-WilcardDNS-hostname",env.EXTERNAL_URL="Your-WilcardDNS-with-https"
```

```bash
Example:
helm install botpress botpress/botpress --version 0.1.1 --set route.host="my-botpress-maximilianopizarro5-dev.apps.sandbox-m2.ll9k.p1.openshiftapps.com",env.EXTERNAL_URL="https://my-botpress-maximilianopizarro5-dev.apps.sandbox-m2.ll9k.p1.openshiftapps.com"
```


## Uninstall Chart

```bash
helm uninstall botpress
```

cc: @redhat-cop/day-in-the-life

ckavili commented:
hey @maximilianoPizarro, thanks for this PR! our linter is a little bit picky 😇 if you can address those new line etc warnings, I'd be happy to merge it 🙏

maximilianoPizarro commented:
> hey @maximilianoPizarro, thanks for this PR! our linter is a little bit picky 😇 if you can address those new line etc warnings, I'd be happy to merge it 🙏

Hi @ckavili! Don´t worry I send the changes on the last commit. Thanks in advance 💯

garethahealy commented:
@maximilianoPizarro ; can you rebase please

maximilianoPizarro commented:
> @maximilianoPizarro ; can you rebase please

Hi @garethahealy, thanks for your help. The rebase is done.

garethahealy commented:
```
charts/botpress/Chart.yaml
  Error: 1:15 [new-lines] wrong new line character: expected \n
  Error: 30:48 [new-line-at-end-of-file] no new line character at the end of file
```

maximilianoPizarro commented:
> ```
> charts/botpress/Chart.yaml
>   Error: 1:15 [new-lines] wrong new line character: expected \n
>   Error: 30:48 [new-line-at-end-of-file] no new line character at the end of file
> ```

@garethahealy  now validation maitainers step fail

 ```
Validation success! 👍
Validating maintainers...
Error: failed linting charts: failed processing charts

------------------------------------------------------------------------------------------------------------------------
 ✖︎ botpress => (version: "0.1.1", path: "charts/botpress") > failed validating maintainer "Maximiliano Pizarro": 404 Not Found
------------------------------------------------------------------------------------------------------------------------
 ```

garethahealy commented on charts/botpress/Chart.yaml:
```diff
@@ -0,0 +1,20 @@
+appVersion: botpress-server-v12
+home: https://maximilianopizarro.github.io/botpress-helm-chart/
+maintainers:
+  - name: Maximiliano Pizarro
```
its wants your github tag here, see: https://github.com/redhat-cop/helm-charts/blob/main/charts/ansible-automation-platform/Chart.yaml#L7-L10

maximilianoPizarro commented:
Thanks for the feedback @garethahealy . I rebased main, and for some reason, the **operator-installe**r wasn't working with my version of main [3.2.1](https://github.com/redhat-cop/helm-charts/blob/9da7d344a318643ab189eecec876a57959005bd9/charts/operators-installer/Chart.yaml#L19), so I updated it to version **3.2.2**, and it works.

garethahealy commented:
@ckavili you happy with everything?
